# 05 Category Flows - Data Cleaning

## Objective

Clean and preprocess the category-wise mutual fund flows dataset before using it for Exploratory Data Analysis (EDA).

The cleaning process includes:

- Loading the raw dataset
- Inspecting dataset structure
- Identifying missing values
- Checking duplicate records
- Correcting data types
- Validating cleaned data
- Exporting the cleaned dataset

In [2]:
import pandas as pd
import numpy as np
from pathlib import Path

In [5]:
category_df = pd.read_csv("../Data/Raw/05_category_inflows.csv")

print("DATASET INFORMATION")


display(category_df.head())

print("\nShape:", category_df.shape)

print("\nColumns:")
print(category_df.columns)

print("\nData Types:")
print(category_df.dtypes)

print("\nMissing Values:")
display(category_df.isnull().sum())

print("\nDuplicate Records:")
print(category_df.duplicated().sum())

DATASET INFORMATION


,month,category,net_inflow_crore
0,2024-04,Large Cap,2413.0
1,2024-04,Mid Cap,3897.0
2,2024-04,Small Cap,3533.0
3,2024-04,Flexi Cap,4947.0
4,2024-04,Large & Mid Cap,4214.0



Shape: (144, 3)

Columns:
Index(['month', 'category', 'net_inflow_crore'], dtype='object')

Data Types:
month                object
category             object
net_inflow_crore    float64
dtype: object

Missing Values:


month               0
category            0
net_inflow_crore    0
dtype: int64


Duplicate Records:
0


In [8]:
# Data Cleaning

# Convert month to datetime
category_df["month"] = pd.to_datetime(category_df["month"])

# Sort chronologically
category_df = category_df.sort_values(["month", "category"]).reset_index(drop=True)

# Verify
category_df.info()

category_df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 144 entries, 0 to 143
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   month             144 non-null    datetime64[ns]
 1   category          144 non-null    object        
 2   net_inflow_crore  144 non-null    float64       
dtypes: datetime64[ns](1), float64(1), object(1)
memory usage: 3.5+ KB


,month,category,net_inflow_crore
0,2024-04-01,ELSS,466.0
1,2024-04-01,Flexi Cap,4947.0
2,2024-04-01,Gilt,784.0
3,2024-04-01,Hybrid,2955.0
4,2024-04-01,Large & Mid Cap,4214.0


In [9]:
print("DATA VALIDATION")


print(f"Shape : {category_df.shape}")
print(f"Duplicate Rows : {category_df.duplicated().sum()}")

print("\nMissing Values")
print(category_df.isnull().sum())

print("\nDate Range")
print(category_df["month"].min())
print(category_df["month"].max())

print("\nUnique Categories :", category_df["category"].nunique())

DATA VALIDATION
Shape : (144, 3)
Duplicate Rows : 0

Missing Values
month               0
category            0
net_inflow_crore    0
dtype: int64

Date Range
2024-04-01 00:00:00
2025-03-01 00:00:00

Unique Categories : 12


In [10]:
output_path = "../Data/Processed/05_category_inflows_cleaned.csv"

category_df.to_csv(output_path, index=False)

print(f"Cleaned dataset saved to:\n{output_path}")

Cleaned dataset saved to:
../Data/Processed/05_category_inflows_cleaned.csv


## Cleaning Summary

### Cleaning Performed

- Converted the `month` column from object to datetime format.
- Sorted the dataset by month and category.
- Verified duplicate records.
- Verified missing values.
- Exported the cleaned dataset to the Processed folder.

### Observations

- No missing values were identified.
- No duplicate records were present.
- The dataset required only a datetime conversion for the `month` column.

### Final Dataset

- Records: 144
- Features: 3
- Date Range: April 2024 – March 2025
- Categories: 12 Mutual Fund Categories